neqo-http3/src/connection.rs

```diff
@@ -796,22 +799,60 @@ impl Http3Connection {
             .as_request_target()
             .map_err(|_| Error::InvalidRequestTarget)?;
 
-        // Transform pseudo-header fields
-        let mut final_headers = vec![
-            Header::new(":method", request.method),
-            Header::new(":scheme", target.scheme()),
-            Header::new(":authority", target.authority()),
-            Header::new(":path", target.path()),
-        ];
-        if let Some(conn_type) = request.connect_type {
-            final_headers.push(Header::new(":protocol", conn_type.string()));
+        match request.connect_type {
+            Some(_) if request.method != "CONNECT" => {
+                qwarn!(
+                    "Method {} with CONNECT type {:?}",
+                    request.method,
+                    request.connect_type
+                );
+                debug_assert!(false);
+            }
+            None if request.method == "CONNECT" => {
+                qwarn!(
+                    "Method {} with CONNECT type {:?}",
```

Comment:
Maybe fix the message here: you just tested that the method is CONNECT and that the connect type is `None`, so the message can be fixed.

Reply:
:+1: done in https://github.com/mozilla/neqo/pull/2886/commits/f2ac6e41421ac21e41b7eeea8ae610f8872fe6a1.